### ISSUE TITLE
Add PuppetStrings(YARD) doc standard for Puppet

### ISSUE THREAD

**Issue by run2cmd** [kkoomen/vim-doge#35]
**Documentation standard**
Puppet Strings

**How popular is the doc standard?**
- [ ] Not popular at all
- [ ] Some people use it
- [ ] Quite a lot of people use it
- [ ] It's not the most popular one, but a good alternative
- [X] Everyone is using it

It is strongly based on YARD: https://github.com/puppetlabs/puppet-strings. This is standard that Puppet code use. I think best examples are here: https://puppet.com/docs/puppet/6.7/puppet_strings_style.html

**Comment by kkoomen** [comment]
@run2cmd Is Puppet considered a framework or a new language? Since it uses ruby syntax and even uses `.rb` files. Can you explain a bit more about these parts?

**Comment by run2cmd** [comment]
Puppet is a  open-core software configuration management tool that follow principle everything as a code. It was written in ruby at the beginning but now it has parts in Clojure and C++ also.

I would treat it as a new language however it is DSL not GPL. Puppet modules does use `.rb` files for Rspec tests, custom addons/plugins/facts, and `.erb` files for templates, however Puppet code files are `.pp` extension. 

Those `.pp` files are those that use Puppet-Strings documentation.

**Comment by kkoomen** [comment]
Do you suggest to use the `.pp` file extension or `.rb`?

**Comment by run2cmd** [comment]
Yes `.pp` it should be. '.rb' Should be treated as Ruby(YARD) which DoGe already supports.
